client/src/main/java/org/eclipse/hono/client/impl/HonoClientImpl.java

```diff
@@ -592,7 +598,17 @@ private void reconnect(
         return result;
     }
 
-    private Future<RequestResponseClient> newCredentialsClient(final String tenantId) {
+    /**
+     * Custom implementation of {@link CredentialsClient} can be instantiated by overriding this
+     * method. Any such instance should be scoped to the given tenantId.
+     *
```

Comment:
We always put a `<p>` after the introduction to separate it from the detailed explanation.